liquibase-standard/src/main/java/liquibase/diff/output/changelog/core/MissingColumnChangeGenerator.java

```diff
@@ -63,9 +66,10 @@ public Change[] fixMissing(DatabaseObject missingObject, DiffOutputControl contr
         AddColumnConfig columnConfig = createAddColumnConfig();
         columnConfig.setName(column.getName());
 
-        String dataType = column.getType().toString();
-
-        columnConfig.setType(dataType);
+        LiquibaseDataType ldt = DataTypeFactory.getInstance().from(column.getType(), referenceDatabase);
```

Comment:
Would it be better to extract this updated code into a new method, so then we can unit-test it? Otherwise, we can only test just the whole thing?